### PR TITLE
flow: add extra layers

### PR DESCRIPTION
**Pull request by safchain**
This add capability to add
extra layers to flows

### CODE REVIEW

**Comment by safchain** [comment]
rerun tests

File: Makefile [hunk 1]
```diff
@@ -190,10 +192,12 @@ debug.analyzer:
 	$(call skydive_debug,analyzer)
 
 %.pb.go: %.proto
-	$(call VENDOR_RUN,${PROTOC_GEN_GO_GITHUB}) protoc --go_out . $<
+	$(call VENDOR_RUN,${PROTOC_GEN_GOFAST_GITHUB})
+	$(call VENDOR_RUN,${PROTOC_GEN_GO_GITHUB}) protoc -Ivendor -I. --plugin=vendor/${PROTOC_GEN_GOFAST_GITHUB}/protoc-gen-gogofaster --gogofaster_out . $<
```
Comment (hunchback) [inline Makefile]: create a define including both these lines and then reuse on 199-200 as well:

```
define PROTOC_SETUP
$(call VENDOR_RUN,${PROTOC_GEN_GOFAST_GITHUB})
$(call VENDOR_RUN,${PROTOC_GEN_GO_GITHUB}) protoc -Ivendor -I. --plugin=vendor/${PROTOC_GEN_GOFAST_GITHUB}/protoc-gen-gogofaster --gogofaster_out . $1
endef
```

**Comment by safchain** [comment]
run skydive-unit-tests

**Comment by safchain** [comment]
run skydive-unit-tests

**Comment by safchain** [comment]
run skydive-cdd-overview-tests

File: api/types/types.go [hunk 2]
```diff
@@ -75,20 +76,21 @@ func NewAlert() *Alert {
...
+	Count          int              `json:"Count"`
+	PCAPSocket     string           `json:"PCAPSocket,omitempty"`
+	Port           int              `json:"Port,omitempty"`
+	RawPacketLimit int              `json:"RawPacketLimit,omitempty" valid:"isValidRawPacketLimit"`
+	HeaderSize     int              `json:"HeaderSize,omitempty" valid:"isValidCaptureHeaderSize"`
+	ExtraTCPMetric bool             `json:"ExtraTCPMetric"`
+	IPDefrag       bool             `json:"IPDefrag"`
+	ReassembleTCP  bool             `json:"ReassembleTCP"`
+	LayerKeyMode   string           `json:"LayerKeyMode,omitempty" valid:"isValidLayerKeyMode"`
+	ExtraLayers    flow.ExtraLayers `json:"ExtraLayers,omitempty"`
```
Comment (lebauce) [inline api/types/types.go]: Please add a validator so we can't specify an invalid layer

**Pull request merged**